nomos-da/network/messages/src/common.rs

```diff
@@ -0,0 +1,42 @@
+use kzgrs_backend::common::blob::DaBlob;
+use nomos_core::da::BlobId;
+use serde::{Deserialize, Serialize};
+
+#[repr(C)]
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct Blob {
+    pub blob_id: BlobId,
+    pub data: DaBlob,
+}
+
+impl Blob {
+    pub fn new(blob_id: BlobId, data: DaBlob) -> Self {
+        Self { blob_id, data }
+    }
+}
+
+#[repr(C)]
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum CloseMessageReason {
+    GracefulShutdown = 0,
+    SubnetChange = 1,
+    SubnetSampleFailure = 2,
+}
```

Comment:
Ref: https://rust-lang.github.io/unsafe-code-guidelines/layout/enums.html#layout-of-a-fieldless-enum
```
The `#[repr(C)]` annotation [...] selects the same size as the C compiler would use for the given target for an equivalent C-enum declaration.
```

Ref: https://doc.rust-lang.org/reference/type-layout.html#reprc-field-less-enums
```
For field-less enums, the C representation has the size and alignment of the default enum size and alignment for the target platform’s C ABI.
```

Reply:
Yeah, I think in this specific case it is better `repr(u8)` as all variants fits in that smaller size. But we still need to comply to c so `#[repr(C, u8)]`

Reply:
```
Combining a C and Int repr (e.g., #[repr(C, u8)]) is not permitted on a fieldless enum.
```

Reply:
> ```
> Combining a C and Int repr (e.g., #[repr(C, u8)]) is not permitted on a fieldless enum.
> ```

Just `repr(u8)` then. It is more compact and since this goes on wire its better.